File: src/passenger-app/lib/models/order.dart
class Order {
  final int id;
  final String orderNo;
  final int passengerId;
  final int? driverId;
  final int status;
  final String pickupAddr;
  final double pickupLat;
  final double pickupLng;
  final String dropoffAddr;
  final double dropoffLat;
  final double dropoffLng;
  final double estPrice;
  final double actualPrice;
  final int estDistance;
  final int estDuration;
  final int carType;
  final DateTime? departureTime;
  final DateTime? assignedAt;
  final String? createdAt;
  final String? acceptedAt;
  final String? arrivedAt;
  final String? startedAt;
  final String? endedAt;
  final String? cancelledAt;
  final String cancelReason;
  final Driver? driver;

  const Order({
    required this.id,
    required this.orderNo,
    required this.passengerId,
    this.driverId,
    required this.status,
    required this.pickupAddr,
    required this.pickupLat,
    required this.pickupLng,
    required this.dropoffAddr,
    required this.dropoffLat,
    required this.dropoffLng,
    required this.estPrice,
    required this.actualPrice,
    required this.estDistance,
    required this.estDuration,
    required this.carType,
    this.departureTime,
    this.assignedAt,
    this.createdAt,
    this.acceptedAt,
    this.arrivedAt,
    this.startedAt,
    this.endedAt,
    this.cancelledAt,
    required this.cancelReason,
    this.driver,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['id'] as int,
      orderNo: json['order_no'] as String,
      passengerId: json['passenger_id'] as int,
      driverId: json['driver_id'] as int?,
      status: json['status'] as int,
      pickupAddr: json['pickup_addr'] as String,
      pickupLat: (json['pickup_lat'] as num).toDouble(),
      pickupLng: (json['pickup_lng'] as num).toDouble(),
      dropoffAddr: json['dropoff_addr'] as String,
      dropoffLat: (json['dropoff_lat'] as num).toDouble(),
      dropoffLng: (json['dropoff_lng'] as num).toDouble(),
      estPrice: (json['est_price'] as num).toDouble(),
      actualPrice: (json['actual_price'] as num).toDouble(),
      estDistance: json['est_distance'] as int,
      estDuration: json['est_duration'] as int,
      carType: json['car_type'] as int,
      departureTime: json['departure_time'] != null
          ? DateTime.parse(json['departure_time'] as String)
          : null,
      assignedAt: json['assigned_at'] != null
          ? DateTime.parse(json['assigned_at'] as String)
          : null,
      createdAt: json['created_at'] as String?,
      acceptedAt: json['accepted_at'] as String?,
      arrivedAt: json['arrived_at'] as String?,
      startedAt: json['started_at'] as String?,
      endedAt: json['ended_at'] as String?,
      cancelledAt: json['cancelled_at'] as String?,
      cancelReason: json['cancel_reason'] as String? ?? '',
      driver: json['driver'] != null ? Driver.fromJson(json['driver']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'order_no': orderNo,
      'passenger_id': passengerId,
      'driver_id': driverId,
      'status': status,
      'pickup_addr': pickupAddr,
      'pickup_lat': pickupLat,
      'pickup_lng': pickupLng,
      'dropoff_addr': dropoffAddr,
      'dropoff_lat': dropoffLat,
      'dropoff_lng': dropoffLng,
      'est_price': estPrice,
      'actual_price': actualPrice,
      'est_distance': estDistance,
      'est_duration': estDuration,
      'car_type': carType,
      if (departureTime != null)
        'departure_time': departureTime!.toIso8601String(),
      if (assignedAt != null)
        'assigned_at': assignedAt!.toIso8601String(),
      'created_at': createdAt,
      'accepted_at': acceptedAt,
      'arrived_at': arrivedAt,
      'started_at': startedAt,
      'ended_at': endedAt,
      'cancelled_at': cancelledAt,
      'cancel_reason': cancelReason,
    };
  }

  String get statusText {
    switch (status) {
      case 1: return '待派单';
      case 2: return '已派单';
      case 3: return '司机已接单';
      case 5: return '司机已到达';
      case 6: return '行程中';
      case 7: return '已完成';
      case 8: return '已取消';
      default: return '未知';
    }
  }
}

class Driver {
  final int id;
  final String? nickname;
  final String? phone;
  final double? rating;
  final Vehicle? vehicle;

  const Driver({
    required this.id,
    this.nickname,
    this.phone,
    this.rating,
    this.vehicle,
  });

  factory Driver.fromJson(Map<String, dynamic> json) {
    return Driver(
      id: json['id'] as int,
      nickname: json['user']?['nickname'] as String?,
      phone: json['user']?['phone'] as String?,
      rating: (json['rating'] as num?)?.toDouble(),
      vehicle: json['vehicle'] != null ? Vehicle.fromJson(json['vehicle']) : null,
    );
  }
}

class Vehicle {
  final int id;
  final String plateNumber;
  final String brand;
  final String model;
  final String color;
  final int carType;

  const Vehicle({
    required this.id,
    required this.plateNumber,
    required this.brand,
    required this.model,
    required this.color,
    required this.carType,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json['id'] as int,
      plateNumber: json['plate_number'] as String,
      brand: json['brand'] as String,
      model: json['model'] as String,
      color: json['color'] as String,
      carType: json['car_type'] as int,
    );
  }
}

File: src/ride-hermes/internal/model/order.go
package model

import "time"

const (
	OrderStatusPending       int8 = 1
	OrderStatusAssigned      int8 = 2
	OrderStatusAccepted      int8 = 3
	OrderStatusWaitingPickup int8 = 5
	OrderStatusInTrip        int8 = 6
	OrderStatusCompleted     int8 = 7
	OrderStatusCancelled     int8 = 8
)

var OrderStatusNames = map[int8]string{
	OrderStatusPending:       "待派单",
	OrderStatusAssigned:      "已派单",
	OrderStatusAccepted:      "司机已确认",
	OrderStatusWaitingPickup: "司机到达等待",
	OrderStatusInTrip:        "行程中",
	OrderStatusCompleted:     "已完成",
	OrderStatusCancelled:     "已取消",
}

type Order struct {
	BaseModel
	OrderNo      string     `gorm:"uniqueIndex;size:32;not null" json:"order_no"`
	PassengerID  int64      `gorm:"index;not null" json:"passenger_id"`
	DriverID     *int64     `gorm:"index" json:"driver_id"`
	Status       int8       `gorm:"index;not null;default:1" json:"status"`
	PickupAddr   string     `gorm:"size:255;not null" json:"pickup_addr"`
	PickupLat    float64    `gorm:"type:decimal(10,7);not null" json:"pickup_lat"`
	PickupLng    float64    `gorm:"type:decimal(10,7);not null" json:"pickup_lng"`
	DropoffAddr  string     `gorm:"size:255;not null" json:"dropoff_addr"`
	DropoffLat   float64    `gorm:"type:decimal(10,7);not null" json:"dropoff_lat"`
	DropoffLng   float64    `gorm:"type:decimal(10,7);not null" json:"dropoff_lng"`
	EstPrice     float64    `gorm:"type:decimal(10,2);not null;default:0" json:"est_price"`
	ActualPrice  float64    `gorm:"type:decimal(10,2);not null;default:0" json:"actual_price"`
	EstDistance  int        `gorm:"not null;default:0" json:"est_distance"`
	EstDuration  int        `gorm:"not null;default:0" json:"est_duration"`
	CarType       int8       `gorm:"not null;default:1" json:"car_type"`
	DepartureTime time.Time  `gorm:"not null" json:"departure_time"`
	AcceptedAt   *time.Time `json:"accepted_at"`
	AssignedAt   *time.Time `json:"assigned_at"`
	ArrivedAt    *time.Time `json:"arrived_at"`
	StartedAt    *time.Time `json:"started_at"`
	EndedAt      *time.Time `json:"ended_at"`
	CancelledAt  *time.Time `json:"cancelled_at"`
	DemandID     *int64     `gorm:"index" json:"demand_id"`
	MatchingMode int8       `gorm:"not null;default:1" json:"matching_mode"` // 1=Platform, 2=A2A
	CancelReason string     `gorm:"size:255;not null;default:''" json:"cancel_reason"`
	Passenger    User       `gorm:"foreignKey:PassengerID" json:"passenger,omitempty"`
	Driver       *Driver    `gorm:"foreignKey:DriverID" json:"driver,omitempty"`
}

func (Order) TableName() string { return "orders" }
